src/tools/networkTools.cpp

```diff
@@ -211,39 +211,39 @@ std::map<std::string, std::string> getNetworkInterfaces() {
   return result;
 }
 
-
-std::string getBestPublicIp(bool ipv6) {
-  IpAddress bestPublicIp = IpAddress{"127.0.0.1","::1"};
+IpAddress getBestPublicIps(IpMode mode) {
+  IpAddress bestPublicIps = IpAddress{"127.0.0.1", "::1"};
   std::map<std::string, IpAddress> interfaces = getNetworkInterfacesIPv4Or6();
-
+  bool validMode4 = (mode == IpMode::ipv4 || mode == IpMode::all);  // Mode is valid to support ipv4
+  bool validMode6 = (mode == IpMode::ipv6 || mode == IpMode::all);  // Mode is valid to support ipv6
```

Comment:
I think that these can be moved to the bottom of the function. The function will consist of two parts
- select the best public IPv4 and IPv6 addresses regardless of the mode requested (the logic of this will be slightly simpler compared to the current code)
- compose the result from those IPv4 and IPv6 addresses based on the requested mode

Reply:
I am debating whether we should do a mode check inside this function at all. The name `getBestPublicIps` I think better suits a behavior where the whole `IpAdress` struct is filled. IMO mode checking should be moved outside this function.

Reply:
Agree. In fact, when writing my previous comment I considered proposing to package the first part (covered by the first bullet) into a separate function `IpAddress getBestPublicIps(void)`. I even think that you proposed something like that earlier.

Reply:
> I even think that you proposed something like that earlier.

Indeed, [you did](https://github.com/kiwix/libkiwix/pull/1132#discussion_r1765135280).

Reply:
Yeah, this has been all over the place, but let me correct it to close this up.